qcodes/tests/dataset/test_database_creation_and_upgrading.py

```diff
@@ -0,0 +1,204 @@
+import os
+from contextlib import contextmanager
+from copy import deepcopy
+import logging
+import tempfile
+
+import pytest
+
+import qcodes as qc
+from qcodes import new_experiment, new_data_set, ParamSpec
+from qcodes.dataset.database import (initialise_database,
+                                     initialise_or_create_database_at)
+# pylint: disable=unused-import
+from qcodes.tests.dataset.temporary_databases import (empty_temp_db,
+                                                      experiment,
+                                                      temporarily_copied_DB)
+from qcodes.dataset.sqlite_base import (connect,
+                                        update_GUIDs,
+                                        get_user_version,
+                                        atomic_transaction,
+                                        perform_db_upgrade_0_to_1)
+
+from qcodes.dataset.guids import parse_guid
+import qcodes.tests.dataset
+
+fixturepath = os.sep.join(qcodes.tests.dataset.__file__.split(os.sep)[:-1])
+fixturepath = os.path.join(fixturepath, 'fixtures')
+
+
+@contextmanager
+def location_and_station_set_to(location: int, work_station: int):
+    cfg = qc.Config()
+    old_cfg = deepcopy(cfg.current_config)
+    cfg['GUID_components']['location'] = location
+    cfg['GUID_components']['work_station'] = work_station
+    cfg.save_to_home()
+
+    yield
```

Comment:
Should this have a try finally around the post actions too

Reply:
Oh, shoot!